spark/v3.4/spark/src/main/java/org/apache/iceberg/spark/functions/DaysFunction.java

```diff
@@ -116,4 +119,27 @@ public Integer produceResult(InternalRow input) {
       return input.isNullAt(0) ? null : invoke(input.getLong(0));
     }
   }
+
+  public static class TimestampNtzToDaysFunction extends BaseToDaysFunction {
+    // magic method used in codegen
+    public static int invoke(long micros) {
+      return DateTimeUtil.microsToDays(micros);
+    }
+
+    @Override
+    public DataType[] inputTypes() {
+      return new DataType[] {DataTypes.TimestampNTZType};
+    }
+
+    @Override
+    public String canonicalName() {
+      return "iceberg.days(timestamp)";
```

Comment:
timestampntz?

Reply:
Good one! I changed it to `timestamp_ntz` (since that's what Spark is also using)